lib/package/plugins/TD006-target-LoadBalancer-SslInfo.js

```diff
@@ -0,0 +1,75 @@
+/*
+  Copyright 2019-2024 Google LLC
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+const ruleId = require("../myUtil.js").getRuleId();
+
+const plugin = {
+  ruleId,
+  name: "TargetEndpoint HTTPTargetConnection SSLInfo with LoadBalancer",
+  message:
+    "TargetEndpoint HTTPTargetConnection should not use SSLInfo with LoadBalancer.",
+  fatal: false,
+  severity: 1, // 1 = warn, 2 = error
+  nodeType: "Endpoint",
+  enabled: true
+};
+
+const path = require("path"),
+  debug = require("debug")("apigeelint:" + ruleId);
+
+const onTargetEndpoint = function (endpoint, cb) {
+  const htc = endpoint.getHTTPTargetConnection(),
+    shortFilename = path.basename(endpoint.getFileName());
+  let flagged = false;
+
+  debug(`onTargetEndpoint shortfile(${shortFilename})`);
+  if (htc) {
+    try {
+      const loadBalancers = htc.select("LoadBalancer");
+      const sslInfos = htc.select("SSLInfo");
+      debug(
+        `sslInfos (${sslInfos.length}) loadBalancers(${loadBalancers.length})`
+      );
+
+      if (loadBalancers.length == 1) {
+        if (sslInfos.length > 0) {
+          endpoint.addMessage({
+            plugin,
+            line: sslInfos[0].lineNumber,
+            column: sslInfos[0].columnNumber,
+            message: `When using a LoadBalancer, the SSLInfo should not be configured under HTTPTargetConnection`
```

Comment:
Should we also mention that it should be included as part of the TargetServer configuration?

I feel we are only telling what is wrong but not suggesting what to do

Probably changing `When using a LoadBalancer, the SSLInfo should not be configured under HTTPTargetConnection` to `When using a LoadBalancer, the SSLInfo should not be configured under HTTPTargetConnection, it should be configured as part of Target Server` or something on that lines

Reply:
I like that suggestion.  I've improved the error message, what do you think?